src/aof.c

```diff
@@ -653,15 +1051,10 @@ void feedAppendOnlyFile(int dictid, robj **argv, int argc) {
     /* Append to the AOF buffer. This will be flushed on disk just before
      * of re-entering the event loop, so before the client will get a
      * positive reply about the operation performed. */
-    if (server.aof_state == AOF_ON)
+    if (server.aof_state == AOF_ON ||
+        (server.aof_state == AOF_WAIT_REWRITE && server.child_pid != -1)) {
         server.aof_buf = sdscatlen(server.aof_buf,buf,sdslen(buf));
-
-    /* If a background append only file rewriting is in progress we want to
-     * accumulate the differences between the child DB and the current one
-     * in a buffer, so that when the child process will do its work we
-     * can append the differences to the new append only file. */
-    if (server.child_type == CHILD_TYPE_AOF)
-        aofRewriteBufferAppend((unsigned char*)buf,sdslen(buf));
+    }
```

Comment:
Yes,  double writing will make the code look more complicated (I implemented it in #9539 ).  

 In addition, although redis may create infinite number of files on the disk if AOFRW is repeatedly failing. But this is almost impossible or very rare in reality, because AOFRW will bring fork overhead every time, and users will not let AOFRW fail like this.

Reply:
can you estimate how much more overhead this feature adds?
i guess the double writing code on it's own is not more than 5 lines.
but the file tracking / renaming may be a bit more...

I do think repeated failures will be common... user can find out he has insufficient memory after a week of repeated failures and maybe thousands of files on the disk.

Reply:
I don’t think there will be a lot of work. The key is that we are sure that it is necessary to do this. I think this will cause two problems:
1. Code understanding may bring some burdens
2. When double writing files (obviously we can't get around the atomicity problem), when one  successfully and another file fails written, should we directly exit?

Of course,  double writing can indeed solve the possibility of hundreds of INCR AOFs.

Therefore, we can make a decision as soon as possible, I can proceed to implement it.

Reply:
@oranagra Do you have any ideas here? I wonder if we still have a solution, that is, once we find that our INCR AOF number reaches a threshold (for example 32, this means AOFRW have  consecutive failed 32 times), we stop the automatic retry AOFRW and print a log and set a global status. I think the current unlimited AOFRW is not very reasonable, because AOFRW will bring fork overhead every time. what do you think?

Reply:
I discussed it with the core team today.
we concluded that we rather not invest in double write for now, and we can always add that later if we get feedback about problems with infinite number of files.

The suggestion of some AOFRW throttling also came up.
i'm not certain how to do it (don't like to change a config), we can try thinking of it, but that's also something that can be added in the future.

Reply:
Well, I don't think this requires a configuration. We can simulate the tcp retransmission algorithm (Binary Exponential Back off). When AOFRW fails, we have at least twice the time (or size) to retry AOFRW again (Don't try again immediately like now). If it fails again, it will double.

This is just my simple idea, maybe we have a better way.

Reply:
ohh, sorry, i was confusing `server.aof_rewrite_base_size` with `server.aof_rewrite_perc` (didn't want the backoff be visible in `CONFIG GET`).